hooks/KEP.js

```diff
@@ -0,0 +1,174 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const debug = require('debug')('signalk-parser-nmea0183/KEP')
+const utils = require('@signalk/nmea0183-utilities')
+
+  /*
+  === PNKEP,01 - NKE Target speed ===
+  ------------------------------------------------------------------------------
+         0   1  2   3 4  5
+         |   |  |   | |  |
+  $PNKEP,01,x.x,N,x.x,K*hh<CR><LF>
+  ------------------------------------------------------------------------------
+  $PNKEP,01,8.3,N,15.5,K*52
+  Field Number:
+  0: NKE Target speed sentence
+  1: Target speed in knots
+  2: N = knots
+  3: Target speed in km/h
+  4: K
+  5: Checksum
+
+  === PNKEP,02 - Course (COG) on next track ===
+  ------------------------------------------------------------------------------
+         0   1   2
+         |   |   |
+  $PNKEP,02,x.x*hh<CR><LF>
+  ------------------------------------------------------------------------------
+  $PNKEP,02,344.4*6B
+  Field Number:
+  0: NKE Course on next tack sentence
+  1: Course on next tack from 0 to 359
+  2: Checksum
+
+  === PNKEP,03 - NKE Opt. VMG angle and performance up and downwind ===
+  ------------------------------------------------------------------------------
+         0   1   2   3  4
+         |   |   |   |  |
+  $PNKEP,03,x.x,x.x,x.x*hh<CR><LF>
+  ------------------------------------------------------------------------------
+  $PNKEP,03,152.0,55.2,67.1*69
+  Field Number:
+  0: NKE Opt. VMG angle
+  1: Opt. VMG angle
+  2: performance upwind from 0 to 99% - ignored
+  3: performance downwind from 0 to 99% - ignored
+  4: Checksum
+   */
+
+function isEmpty(mixed) {
+  return ((typeof mixed !== 'string' && typeof mixed !== 'number') || (typeof mixed === 'string' && mixed.trim() === ''))
+}
+
+module.exports = function (parser, input) {
+  try {
+    const {
+      id,
+      sentence,
+      parts,
+      tags
+    } = input
+
+      var delta
+      //PNKEP,01
+
+
+      if (parts[0] === '01') {
+
+        if (parts[1] === '' && parts[3] === '') {
+          return Promise.resolve(null)
+        }
+
+        let polarspeed = 0.0
+
+          if (utils.float(parts[3]) > 0 && String(parts[4]).toUpperCase() === 'K') {
+            polarspeed = utils.transform(utils.float(parts[3]), 'kph', 'ms');
+          }
+
+          if (utils.float(parts[1]) > 0 && String(parts[2]).toUpperCase() === 'N') {
+            polarspeed = utils.transform(utils.float(parts[1]), 'knots', 'ms');
+          }
+
+          delta = {
+          updates: [{
+              source: tags.source,
+              timestamp: tags.timestamp,
+              values: [{
+                  path: 'performance.polarSpeed',
```

Comment:
The way I read NKE documentation is that they use the definition target boat speed like Silva/Nexus/Garmin do. So the instrument shows a value for a target speed, and  if you are sailing faster, you are sailing too low to the wind (when beating) etc. The SK definition of `performance.targetSpeed` was created for that purpose. `performance.polarSpeed` however is how fast should you sail in these conditions and this relative wind angle.

Reply:
Thanks for looking into that part. I agree with the definition. I built the function to use the polar speed sent out from Qtvlm through the PNKEP,01 sentence. It might be that Qtvlm (or I for that matter) has misinterpreted the documentation. Can you point me to where it is? If there is something wrong I should fix it here and ask Philippe, the author of Qtvlm, to change on his side instead.

Reply:
I found info in a NKE manual. http://nke-marine-electronics.fr/wp-content/uploads/2017/01/TL-25-UK.pdf 
$PNKEP,01 Target speed VIT_CIBLE.
$PNKEP,02 Heading at next board CAP_AUTRE_BORD.
$PNKEP,03 Optimum upwind angle ANGLE_OPT_VENT, REND_PRES, REND_POLAIRE.
$PNKEP,04 Angles to optimise the CMG and VMG and gain ANGLE_OPT_CMG, ANGLE_OPT_VMG,
GAIN_ROUTE_CMG, GAIN_ROUTE_VMG.
$PNKEP,05 Current direction and speed DIREC_COURANT, VITES_COURANT. 

So it should be targetSpeed in signalk. 
I was a bit confused as Qtvlm is a bit ambivalent in the documentation and talks about both target speed and polar speed sent out to NKE.

Reply:
I remember now that I decided to use performance.polarSpeed as this is the conversion made by the signalk-to-nmea0183 converter. 
    https://github.com/SignalK/signalk-to-nmea0183/blob/c8a1cc1a15d76605b94ed5df3ae055e26ebd398e/sentences/PNKEP01.js#L13

Reply:
I see. That one is probably wrong too... It corresponds to $PSILTBS which is better documented. I added that to https://github.com/SignalK/signalk-to-nmea0183/pull/1 which seems to have been misplaced.

Reply:
Ok, so I change to targetSpeed. Thanks.